bundle.py

```diff
@@ -0,0 +1,194 @@
+import configparser
+import os
+import re
+import shutil
+import subprocess
+import sys
+import time
+
+import tomlkit
+
+APP = 'napari'
+WINDOWS = os.name == 'nt'
+MACOS = sys.platform == 'darwin'
+LINUX = sys.platform.startswith("linux")
+HERE = os.path.abspath(os.path.dirname(__file__))
+PYPROJECT_TOML = os.path.join(HERE, 'pyproject.toml')
+SETUP_CFG = os.path.join(HERE, 'setup.cfg')
+
+if WINDOWS:
+    BUILD_DIR = os.path.join(HERE, 'windows')
+elif LINUX:
+    BUILD_DIR = os.path.join(HERE, 'linux')
+elif MACOS:
+    BUILD_DIR = os.path.join(HERE, 'macOS')
+    APP_DIR = os.path.join(BUILD_DIR, APP, f'{APP}.app')
+
+
+with open(PYPROJECT_TOML, 'r') as f:
+    original_toml = f.read()
+
+with open(os.path.join(HERE, "napari", "_version.py")) as f:
+    match = re.search(r'version\s?=\s?\'([^\']+)', f.read())
+    if match:
+        VERSION = match.groups()[0].split('+')[0]
+
+
+def patch_toml():
```

Comment:
for you to consider, I would think patching of the toml file is less explicit, I personally would rather just create a pyproject_bundle.toml

Reply:
I'm not opposed to that, I was just trying to avoid a second place where we have to keep our dependencies synchronized.  (I suppose we could write a test for that...)
It's also quite possible that we will specifically _want_ to have a seperate list of dependencies for the bundle (e.g. to include dependencies that might be annoying for end-users to compile), in which case I would definitely be in favor of a new file.  anyone else want to weigh in on that?

Reply:
> anyone else want to weigh in on that?

rather ... if anyone is opposed to that, speak up ... otherwise I'll go in that direction.

Reply:
I think it's a good idea to have a new file. It would be ideal if the new file could reference the standard file (ie declare that this is everything in pyproject.toml PLUS some bonus goodies), but that might not be feasible, in which case, let's just do the easy duplication. One idea would be to add a CI check that does a set intersection of the dependencies in both files and errors out if the bundle file is not a superset of the base file...

Reply:
updated.
I can't actually make a new `pyproject_bundle.toml` file, since `pyproject.toml` is a special file that briefcase is looking for, so we'd still need to patch the main pyproject at runtime.  So I just populated the `[tool.briefcase.app.napari][requires]` section of `pyproject.toml` with a copy of what's in setup.cfg (plus a couple extras, including zarr) and added a superset test.  I also forgot that `patch_toml` is updating the version of the bundle as well... and I don't want to have to remember to test/update that, so I'm going to leave the version updating part.

Reply:
I was thinking to have one overwrite step during build to replace the file, either approach sounds reasonable to me.